include/zephyr/devicetree.h

```diff
@@ -2237,6 +2268,22 @@
 #define DT_REG_ADDR_BY_NAME(node_id, name) \
 	DT_CAT4(node_id, _REG_NAME_, name, _VAL_ADDRESS)
 
+/**
+ * @brief 64-bit version of a DT_REG_ADDR_BY_NAME macro
+ *
+ * Original DT_REG_ADDR_BY_NAME macro will result in warnings with 64-bit
+ * addresses from devicetree, so this macro should be used
+ * in such cases instead. It's also safe to use it with 32-bit addresses.
```

Comment:
but if we specify that `it's also safe to use it with 32-bit addresses` the user can wonder why we need the 32-bit version at all. Can you specify here also the limitation of this 64-bit version? And possibly be more pedantic when to use one or the other version? Otherwise this is going to confuse a lot of people.

Reply:
Hi @carlocaione! You're right. 
We basically can use _U64 versions instead of the original ones. 
The only reason why we don't replace the original macro is that we can't use _U64 versions inside linker scripts since ld complains on "ULL"

I can rephrase it like this:

```
 * Original AAA macro will result in warnings with 64-bit
 * addresses from devicetree, so this macro should be used
 * in such cases instead. 
 * Generally this macro could be used interchangeably with 
 * AAA macro with one exception: it couldn't be used in linker scripts.
 * The general rule of thumb is to use 64-bit version when the
 * appearance of 64-bit address is possible and when it's not 
 * conflicting with linker scripts
```

What do you say?

Reply:
so, IIUC the only difference is that the `_U64` version is adding the `ULL` suffix? In that case this must be explicit in the description.

Reply:
Ok, thanks. How about this:

```
 * This macro adds ULL postfix to the address value returned
 * by AAA macro. 
 * Generally AAA_U64 macro could be used interchangeably with 
 * AAA macro with one exception: it couldn't be used in linker scripts.
 * The general rule of thumb is to use 64-bit version when the
 * appearance of 64-bit address is possible and when it's not 
 * conflicting with linker scripts
```

Reply:
yes, this is better, thanks.